### PR TITLE
Add unique rules

### PR DESCRIPTION
**Pull request by AMMayberry1**
Implemented the rules for enforcing unique cards in play. Added tests for the edge cases of Yularen and Snoke to confirm that they work correctly.

Yularen worked out of the box. Snoke required adding some additional logic to disable ongoing effects when a card has a pending defeat. Also, refactored the "pending defeat" code to be clearer and operate more consistently.

### CODE REVIEW

File: test/server/cards/01_SOR/units/AgentKallusSeekingTheRebels.spec.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+                // CASE 2: non-unique is defeated, no trigger
+                context.player1.clickCard(context.battlefieldMarine);
+                context.player1.clickCard(context.wampa);
+                expect(context.battlefieldMarine).toBeInLocation('discard');
+                expect(context.player1.handSize).toBe(round1HandSize);
+
+                reset(false);
+
+                // CASE 3: option to draw card when enemy unique unit is defeated (do take)
+                context.player2.clickCard(context.admiralOzzel);
```
Comment (VJubert) [inline test/server/cards/01_SOR/units/AgentKallusSeekingTheRebels.spec.ts:53]: maybe take someone else than ozzel ? when he will be implemented there will be a new step to add here
